sklearn/cross_decomposition/_pls.py

```diff
@@ -295,6 +295,18 @@ def fit(self, X, Y):
 
         self.coef_ = np.dot(self.x_rotations_, self.y_loadings_.T)
         self.coef_ = self.coef_ * self.y_std_
+
+        if self.deflation_mode == "regression":
+            # Calculate VIP for PLSRegression
+            T = self.x_scores_
+            W = self.x_weights_
```

Comment:
Afaik, this should be `W = self.x_rotations_` instead, see [here](https://github.com/scikit-learn/scikit-learn/issues/7050#issuecomment-1399608511).